siatest/renter.go

```diff
@@ -59,17 +59,37 @@ func (tn *TestNode) DownloadByStream(rf *RemoteFile) (data []byte, err error) {
 
 // DownloadInfo returns the DownloadInfo struct of a file. If it returns nil,
 // the download has either finished, or was never started in the first place.
-func (tn *TestNode) DownloadInfo(lf *LocalFile, rf *RemoteFile) (*api.DownloadInfo, error) {
+// If the corresponding download info was found, DownloadInfo also performs a
+// few sanity checks on its fields.
+func (tn *TestNode) DownloadInfo(lf *LocalFile, rf *RemoteFile) (di *api.DownloadInfo, err error) {
 	rdq, err := tn.RenterDownloadsGet()
 	if err != nil {
 		return nil, err
 	}
 	for _, d := range rdq.Downloads {
 		if rf.siaPath == d.SiaPath && lf.path == d.Destination {
-			return &d, nil
+			di = &d
+			break
 		}
 	}
-	return nil, nil
+	if di == nil {
+		// No download info found.
+		return
```

Comment:
I believe that results in returning 'nil', and 'nil' - shouldn't this return an error (info not found or something)? If you return an error to a function that expects the object to exist, it's going to panic.